apps/src/templates/teacherDashboard/SectionRow.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { Component, PropTypes } from 'react';
+import i18n from '@cdo/locale';
+import ProgressButton from '@cdo/apps/templates/progress/ProgressButton';
+
+export const sectionShape = PropTypes.shape({
+  id: PropTypes.number.isRequired,
+  name: PropTypes.string.isRequired,
+  loginType: PropTypes.oneOf(['word', 'email', 'picture']).isRequired,
+  stageExtras: PropTypes.bool.isRequired,
+  pairingAllowed: PropTypes.bool.isRequired,
+  numStudents: PropTypes.number.isRequired,
+  code: PropTypes.string.isRequired,
+  assignmentName: PropTypes.string,
+  assignmentPath: PropTypes.string
+});
+
+const styles = {
+  sectionName: {
+    fontSize: 18,
+    paddingTop: 12
+  },
+};
+
+export default class SectionRow extends Component {
+  static propTypes = {
+    section: sectionShape.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.onClickDelete = this.onClickDelete.bind(this);
+    this.onClickDeleteNo = this.onClickDeleteNo.bind(this);
+    this.onClickDeleteYes = this.onClickDeleteYes.bind(this);
+
+    this.state = {
+      editing: false,
+      deleting: false
+    };
+  }
+
+  onClickDelete() {
+    this.setState({deleting: true});
+  }
+
+  onClickDeleteNo() {
+    this.setState({deleting: false});
+  }
+
+  onClickDeleteYes() {
+    console.log('this is where our delete will happen');
+  }
+
+  render() {
+    const { section } = this.props;
+    return (
+      <tr>
+        <td>
+          <span style={styles.sectionName}>
+            <a href={`#/sections/${section.id}/`}>
+              {section.name}
+            </a>
+          </span>
+        </td>
+        <td>
+          {section.loginType}
+        </td>
+        <td>
+          {section.grade}
+        </td>
+        <td>
+          {section.assignmentName &&
+            <a href={section.assignmentPath}>
+              {section.assignmentName}
+            </a>
+          }
+        </td>
+        <td>
+          {section.stageExtras ? i18n.yes() : i18n.no()}
+        </td>
+        <td>
+          {section.pairingAllowed ? i18n.yes() : i18n.no()}
+        </td>
+        <td>
+          <a href={`#/sections/${section.id}/manage`}>
+            {section.numStudents}
+          </a>
+        </td>
+        <td>
+          {section.code}
+        </td>
+        <td>
+          {/*TODO: i18n */}
+          {!this.state.deleting && (
+            <div style={{whiteSpace: 'nowrap'}}>
+              <ProgressButton
+                text={"Edit"}
+                onClick={() => console.log('start editing')}
+                color={ProgressButton.ButtonColor.gray}
+              />
+              {section.numStudents > 0 && (
+                <ProgressButton
+                  style={{marginLeft: 5}}
+                  text={"Delete"}
+                  onClick={this.onClickDelete}
+                  color={ProgressButton.ButtonColor.red}
+                />
+              )}
+            </div>
+          )}
+          {this.state.deleting && (
```

Comment:
It might be nice to extract the buttons here into a functional stateless component in this file.  The `{!this.state.deleting &&` and `{this.state.deleting &&` side-by-side subtly suggest four states to me, where you could use a proper if/else in a subcomponent.  It also seems likely that this will get more complex as we add the editing states, could extract it then.